### ISSUE TITLE
tools: GOBIN is misused in dlv-dap.exe installation

### ISSUE THREAD

**Issue by wtask** [golang/vscode-go#1430]
Version: 1.55.2 (user setup)
Commit: 3c4e3df9e89829dce27b7b5c24508306b151f30d
Date: 2021-04-13T09:35:57.887Z
Electron: 11.3.0
Chrome: 87.0.4280.141
Node.js: 12.18.3
V8: 8.7.220.31-electron.0
OS: Windows_NT x64 10.0.19042

vscode-go 0.24.1

go env
```
set GO111MODULE=on
set GOARCH=amd64
set GOBIN=c:\dev\go\bin
set GOCACHE=C:\Users\webta\AppData\Local\go-build
set GOENV=C:\Users\webta\AppData\Roaming\go\env
set GOEXE=.exe
set GOFLAGS=
set GOHOSTARCH=amd64
set GOHOSTOS=windows
set GOINSECURE=
set GOMODCACHE=C:\dev\go\pkg\mod
set GONOPROXY=XXX
set GONOSUMDB=XXX
set GOOS=windows
set GOPATH=C:\dev\go
set GOPRIVATE=XXX
set GOPROXY=https://proxy.golang.org,direct
set GOROOT=C:\Program Files\Go
set GOSUMDB=sum.golang.org
set GOTMPDIR=
set GOTOOLDIR=C:\Program Files\Go\pkg\tool\windows_amd64   
set GOVCS=
set GOVERSION=go1.16.3
set GCCGO=gccgo
set AR=ar
set CC=gcc
set CXX=g++
set CGO_ENABLED=1
set GOMOD=YYY\go.mod
set CGO_CFLAGS=-g -O2
set CGO_CPPFLAGS=
set CGO_CXXFLAGS=-g -O2
set CGO_FFLAGS=-g -O2
set CGO_LDFLAGS=-g -O2
set PKG_CONFIG=pkg-config
set GOGCCFLAGS=-m64 -mthreads -fmessage-length=0 -fdebug-prefix-map=C:\Users\webta\AppData\Local\Temp\go-build4034536518=/tmp/go-build -gno-record-gcc-switches
```
I've tried to use dlv-dap for the first time and added following object into settings:
```
  "go.delveConfig": {
    "dlvLoadConfig": {
      "followPointers": true,
      "maxVariableRecurse": 1,
      "maxStringLen": 64,
      "maxArrayValues": 64,
      "maxStructFields": -1
    },
    "apiVersion": 2,
    "showGlobalVariables": false,
    "debugAdapter": "dlv-dap"
  },
```
Than I've reinstalled dlv and dlv-dap with "Go: Install/Update tools" (OK)
And started my debug session, but got an error means that dlv-dap.exe was not found.
I checked my GOBIN folder and right, the debug adapter was saved in GOBIN/bin/dlv-dap.exe.
Moving dlv-dap.exe into GOBIN did resolve an error. Why vscode-go installs dlv-dap into GOBIN subfolder?

**Comment by hyangah** [comment]
Released with v0.24.2 now. @wtask I did a quick check before releasing but if the problem still exists, please let us know.
Thanks again for the report!

And, let us know how your experience with the dlv-dap is!

**Comment by wtask** [comment]
@hyangah I've reinstalled `dlv` and `dlv-dap` just now with no issue. Thank you.
